src/main/java/hudson/plugins/powershell/PowerShell.java

```diff
@@ -15,27 +15,45 @@
  * @author Kohsuke Kawaguchi
  */
 public class PowerShell extends CommandInterpreter {
+
+    boolean stopOnError;
+
     @DataBoundConstructor
-    public PowerShell(String command) {
+    public PowerShell(String command, boolean stopOnError) {
         super(command);
+        this.stopOnError = stopOnError;
+    }
+
+    public boolean isStopOnError() {
+        return stopOnError;
     }
 
     protected String getFileExtension() {
         return ".ps1";
     }
-
+    
     public String[] buildCommandLine(FilePath script) {
         if (isRunningOnWindows(script)) {
             return new String[] { "powershell.exe", "-NonInteractive", "-ExecutionPolicy", "Bypass", "& \'" + script.getRemote() + "\'"};
         } else {
             // ExecutionPolicy option does not work (and is not required) for non-Windows platforms
             // See https://github.com/PowerShell/PowerShell/issues/2742
-            return new String[] { "powershell", "-NonInteractive", "& \'" + script.getRemote() + "\'"};
+            return new String[] { "powershell", "-NonInteractive", "-File", script.getRemote()};
```

Comment:
Is there a reason you made this change?

Unfortunately it changes the behaviour of older versions of PowerShell (<= 4.0) such that uncaught exceptions will no longer exit with a non-zero exit code, which will no longer fail the build.

Try running both of these in PowerShell 4.0 or earlier to see:
```
$tempFile = "$([IO.Path]::GetTempPath())throw.ps1"
"throw 'some error'" > $tempFile
powershell "& $tempFile"
$LastExitCode
```
> Results in `1`

```
$tempFile = "$([IO.Path]::GetTempPath())throw.ps1"
"throw 'some error'" > $tempFile
powershell -File $tempFile
$LastExitCode
```
> Results in `0`

Reply:
This line is for Linux only. I believe only powershell version 6 is available for Linux. I tested with PSVersion 6.0.0-beta.8 and got this error:

```
$ echo "Write-Host 'Hello World!'"  > script.ps1
$ powershell -nonInteractive "& script.ps1"
The argument '& script.ps1' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
$ powershell -nonInteractive -File script.ps1
Hello World!
```

Reply:
Good point. Can you add a comment above the windows invocation to explain that it should use `&` instead of `-File` so that someone doesn't come along and change that to match this line in the future?

Fun fact: I had tested this examples on Powershell 6.0.0-alpha and it executed just fine. Powershell 6.0.0-beta behaves differently!

Reply:
comment added as requested

Reply:
As of this writing, the PowerShell Core executable is now named `pwsh` (Unix), `pwsh.exe` (Windows). 

Also, you just need to use `-c` (`-Command`) in order to pass a piece of PowerShell code rather than a script file.

Windows PowerShell still defaults to `-Command`, but in order to support shebang lines on Unix platforms the default for PowerShell Core was changed to `-File`.

`pwsh -h` shows CLI help.

Reply:
@mklement0  learned something new about powershell/pwsh. But, I think, that should be a separate issue and pull request.

Reply:
@froque: 

I see a need for a separate PR only to perhaps address the desire to optionally use PowerShell _Core_ even on _Windows_.

This PR needs fixing now anyway, and based on the above it looks like all you need is this, which would also address @chrisalbrecht's concern:

```java
return new String[] { "pwsh", "-NonInteractive", "-Command", "& \'" + script.getRemote() + "\'"};
```